src/orchestrator-process/src/lib.rs

```diff
@@ -525,6 +528,20 @@ impl NamespacedProcessOrchestrator {
                 }
             }
 
+            // Clean up scratch directory when the service is terminated.
+            // This is best effort as a development and testing convenience.
+            // Because the process orchestrator is not used in production, we
+            // don't need to be perfectly robust with the cleanup.
+            let _guard = scopeguard::guard((), |_| {
+                if let Some(scratch) = scratch_directory {
+                    info!(scratch_dir = %scratch.display(), "cleaning up scratch directory");
+                    task::spawn(
+                        || "clean_cluster_scratch_directory",
+                        remove_dir_all(scratch),
```

Comment:
you'll still need the `if Let Err(e)` with the println here! Unless someone in review said not to log and I missed it...

Reply:
Yeah, noticed it as well 😅 . Adding it back (I was trying to figure out how to map the future cleanly).

Reply:
Added it back, using `map_err`